securitycenter/notificationsv2/notifications_test.go

```diff
@@ -0,0 +1,265 @@
+		configID := "go-test-delete-config-id" + rand.String()
+
+		if err := addNotificationConfig(t, configID); err != nil {
+			r.Errorf("Could not setup test environment: %v", err)
+			return
+		}
+
+		if err := deleteNotificationConfig(buf, orgID(t), configID); err != nil {
+			r.Errorf("deleteNotificationConfig failed: %v", err)
+			return
+		}
+
+		if !strings.Contains(buf.String(), "Deleted config:") {
+			r.Errorf("deleteNotificationConfig did not delete.")
+		}
+	})
+}
+
+func TestGetNotificationConfig(t *testing.T) {
+	testutil.Retry(t, 5, 30*time.Second, func(r *testutil.R) {
+		buf := new(bytes.Buffer)
+		rand, err := uuid.NewUUID()
+		if err != nil {
+			r.Errorf("Issue generating id.")
+			return
+		}
+		configID := "go-test-get-config-id" + rand.String()
+
+		if err := addNotificationConfig(t, configID); err != nil {
+			r.Errorf("Could not setup test environment: %v", err)
+			return
+		}
+
+		if err := getNotificationConfig(buf, orgID(t), configID); err != nil {
+			r.Errorf("getNotificationConfig failed: %v", err)
+			return
+		}
+
+		if !strings.Contains(buf.String(), "Received config:") {
+			r.Errorf("getNotificationConfig did not delete.")
+		}
+
+		cleanupNotificationConfig(t, configID)
+	})
+}
+
+func TestListNotificationConfigs(t *testing.T) {
+	testutil.Retry(t, 5, 30*time.Second, func(r *testutil.R) {
+		buf := new(bytes.Buffer)
+		rand, err := uuid.NewUUID()
+		if err != nil {
+			r.Errorf("Issue generating id.")
+			return
+		}
+		configID := "go-test-list-config-id" + rand.String()
+
+		if err := addNotificationConfig(t, configID); err != nil {
+			r.Errorf("Could not setup test environment: %v", err)
+			return
+		}
+
+		if err := listNotificationConfigs(buf, orgID(t)); err != nil {
+			r.Errorf("listNotificationConfig failed: %v", err)
+			return
+		}
+
+		if !strings.Contains(buf.String(), "NotificationConfig") {
+			r.Errorf("listNotificationConfigs did not list")
+		}
+
+		cleanupNotificationConfig(t, configID)
+	})
+}
+
+func TestUpdateNotificationConfig(t *testing.T) {
+	testutil.Retry(t, 5, 30*time.Second, func(r *testutil.R) {
+		buf := new(bytes.Buffer)
+		rand, err := uuid.NewUUID()
+		if err != nil {
+			r.Errorf("Issue generating id.")
+			return
+		}
+		configID := "go-test-update-config-id" + rand.String()
+
+		if err := addNotificationConfig(t, configID); err != nil {
+			r.Errorf("Could not setup test environment: %v", err)
+			return
+		}
+
+		if err := updateNotificationConfig(buf, orgID(t), configID, pubsubTopic(t)); err != nil {
+			r.Errorf("updateNotificationConfig failed: %v", err)
+			return
+		}
+
+		if !strings.Contains(buf.String(), "Updated NotificationConfig:") {
+			r.Errorf("updateNotificationConfig did not update.")
+		}
+		cleanupNotificationConfig(t, configID)
+	})
+}
+
+func TestReceiveNotifications(t *testing.T) {
+	testutil.Retry(t, 5, 30*time.Second, func(r *testutil.R) {
+		buf := new(bytes.Buffer)
+		if err := receiveMessages(buf, projectID(t), pubsubSubscription(t)); err != nil {
```

Comment:
this does not look like we're sending a notification. what notifications do you expect this test to receive?

Reply:
Addressed. Updated the functionality